main.m

```diff
@@ -166,55 +300,99 @@ void dumpKeychainEntitlements() {
 	}
-	else if (kSecClassType == kSecClassKey) {
-		return @"No Key Keychain items found.\n";	
+	else if (kSecClassType == kSecClassKey)
+    {
+		return @"[INFO] No Key Keychain items found.\n[HINT] You should unlock your device!\n";	
 	}
-	else {
-		return @"Unknown Security Class\n";
+	else
+    {
+		return @"[INFO] Unknown Security Class\n[HINT] You should unlock your device!\n";
 	}
-	
 }
 
-void printGenericPassword(NSDictionary *passwordItem) {
-	printToStdOut(@"Generic Password\n");
+// Function to convert make the accessible attribute more readable, added protection level to quickly identify weak 
+// protected keychain items.
+void printAccessibleAttribute(NSString *accessibleString)
+{
+	if ([accessibleString isEqualToString:@"dk"]) 
+		printToStdOut(@"%sAccessible Attribute: kSecAttrAccessibleAlways, protection level 0\n%s", KRED, KWHT);
+	else if ([accessibleString isEqualToString:@"ak"])
+		printToStdOut(@"%sAccessible Attribute: kSecAttrAccessibleWhenUnlocked, protection level 2 (default)\n%s", KYEL, KWHT);
+	else if ([accessibleString isEqualToString:@"ck"])
+		printToStdOut(@"%sAccessible Attribute: kSecAttrAccessibleAfterFirstUnlock, protection level 1\n%s", KRED, KWHT);
+	else if ([accessibleString isEqualToString:@"dku"])
+		printToStdOut(@"%sAccessible Attribute: kSecAttrAccessibleAlwaysThisDeviceOnly, protection level 3\n%s", KBLU, KWHT);
+	else if ([accessibleString isEqualToString:@"aku"])
+		printToStdOut(@"%sAccessible Attribute: kSecAttrAccessibleWhenUnlockedThisDeviceOnly, protection level 5\n%s", KBLU, KWHT);
+	else if ([accessibleString isEqualToString:@"cku"])
+		printToStdOut(@"%sAccessible Attribute: kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly, protection level 4\n%s", KBLU, KWHT);
+	else if ([accessibleString isEqualToString:@"akpu"])
+		printToStdOut(@"%sAccessible Attribute: kSecAttrAccessibleWhenPasscodeSetThisDeviceOnly, protection level 6\n%s",KGRN, KWHT);
+	else
+		printToStdOut(@"%sUnknown Accessible Attribute: %@\n%s", KRED, accessibleString, KWHT);
+}
+
+void printGenericPassword(NSDictionary *passwordItem)
+{
+    printToStdOut(@"Generic Password\n");
 	printToStdOut(@"----------------\n");
 	printToStdOut(@"Service: %@\n", [passwordItem objectForKey:(id)kSecAttrService]);
 	printToStdOut(@"Account: %@\n", [passwordItem objectForKey:(id)kSecAttrAccount]);
 	printToStdOut(@"Entitlement Group: %@\n", [passwordItem objectForKey:(id)kSecAttrAccessGroup]);
 	printToStdOut(@"Label: %@\n", [passwordItem objectForKey:(id)kSecAttrLabel]);
-	printToStdOut(@"Generic Field: %@\n", [[passwordItem objectForKey:(id)kSecAttrGeneric] description]);
+	// Added the accessible attribute printout
+	NSString* accessibleString = [passwordItem objectForKey:(id)kSecAttrAccessible];
+	printAccessibleAttribute(accessibleString);
+	// Added the created/mmodified attribute printout, not finaly implemented yet
+	//printToStdOut(@"Creation Date: %@\n", [passwordItem objectForKey:(id)kSecAttrCreationDate]); //CFDate
```

Comment:
Maybe remove the unimplemented things.

Reply:
It could be removed, but maybe someone wants this information.

Reply:
What does "not finaly implemented yet" mean? Is this not available from Apple or would this require more work on our end?

Reply:
It would require more work on our end, I guess it's totally possible to present this information. I don't have a use-case yet, but maybe someone else does.

Reply:
I removed the unimplemented things.